test/models/test_cifar.py

```diff
@@ -32,8 +32,8 @@ def train(X):
     for _ in range(3): train(Tensor.randn(32, 3, 32, 32))
 
     print(f"used {GlobalCounters.mem_used/1e9} GB and {len(train.jit_cache)} kernels")
-    assert GlobalCounters.mem_used/1e9 < 0.55, "CIFAR training used more than 0.55 GB"
-    assert len(train.jit_cache) <= 236, "CIFAR training used more than 236 kernels"
+    assert GlobalCounters.mem_used/1e9 < 0.56, "CIFAR training used more than 0.56 GB"
```

Comment:
This number does increase a tiny bit from intermediate buffers, but before this fix it was sitting at 0.88 GB which was symptomatic of the main issue.

Reply:
Small increase is fine, particularly if the runtime is better.